src/commands/external.rs

```diff
@@ -123,6 +88,98 @@ pub async fn execute_external_subcommand(
     Ok(())
 }
 
+async fn ensure_plugin_available(
+    plugin_name: &str,
+    plugin_store: &PluginStore,
+    app: clap::App<'_>,
+    override_compatibility_check: bool,
+) -> anyhow::Result<Option<String>> {
+    let plugin_version = match plugin_store.read_plugin_manifest(plugin_name) {
+        Ok(manifest) => {
+            if let Err(e) =
+                warn_unsupported_version(&manifest, SPIN_VERSION, override_compatibility_check)
+            {
+                eprintln!("{e}");
+                // TODO: consider running the update checked?
+                process::exit(1);
+            }
+            Some(manifest.version().to_owned())
+        }
+        Err(PluginError::NotFound(e)) => {
+            consider_install(plugin_name, plugin_store, app, &e).await?
+        }
+        Err(e) => return Err(e.into()),
+    };
+    Ok(plugin_version)
+}
+
+async fn consider_install(
+    plugin_name: &str,
+    plugin_store: &PluginStore,
+    app: clap::App<'_>,
+    e: &spin_plugins::error::NotFoundError,
+) -> anyhow::Result<Option<String>> {
+    if predefined_externals()
+        .iter()
+        .any(|(name, _)| name == plugin_name)
+    {
+        println!("The `{plugin_name}` plugin is required. Installing now.");
+        let plugin_installer = installer_for(plugin_name);
+        // Automatically update plugins if the cloud plugin manifest does not exist
+        // TODO: remove this eventually once very unlikely to not have updated
+        if let Err(e) = plugin_installer.run().await {
+            if let Some(PluginError::NotFound(_)) = e.downcast_ref::<PluginError>() {
+                update().await?;
+            }
+            plugin_installer.run().await?;
+        }
+        Ok(None) // No update badgering needed if we just updated/installed it!
+    } else if stderr().is_terminal() && matches_catalogue_plugin(plugin_store, plugin_name) {
+        if offer_install(plugin_name)? {
+            let plugin_installer = installer_for(plugin_name);
+            plugin_installer.run().await?;
+            eprintln!();
+            Ok(None) // No update badgering needed if we just updated/installed it!
+        } else {
+            process::exit(2);
+        }
+    } else {
+        tracing::debug!("Tried to resolve {plugin_name} to plugin, got {e}");
+        terminal::error!("'{plugin_name}' is not a known Spin command. See spin --help.\n");
+        print_similar_commands(app, plugin_name);
+        process::exit(2);
+    }
+}
+
+fn offer_install(name: &str) -> anyhow::Result<bool, anyhow::Error> {
+    terminal::warn!("`{name}` is not a known Spin command, but there is a plugin with that name.");
```

Comment:
Alluding to your thought in the PR description; perhaps printing the plugin description around here would be useful to the user.  Link to repo as well?  (Or getting to crowded at this point?)

Reply:
Yes, that was what was on my mind - give them a bit more info to make the decision.  But for now I think I will just merge this and we can iterate on the description thing separately.